File: src/test/java/net/bramp/ffmpeg/fixtures/Progresses.java
package net.bramp.ffmpeg.fixtures;

import com.google.common.collect.ImmutableList;
import net.bramp.ffmpeg.progress.Progress;

public final class Progresses {

  private Progresses() {
    throw new AssertionError("No instances for you!");
  }

  public static final ImmutableList<String> allFiles =
      ImmutableList.of("ffmpeg-progress-0", "ffmpeg-progress-1", "ffmpeg-progress-2");

  public static final ImmutableList<String> naProgressFile = ImmutableList.of("ffmpeg-progress-na");

  public static final ImmutableList<Progress> allProgresses =
      ImmutableList.of(
          new Progress(
              5,
              org.apache.commons.lang3.math.Fraction.getFraction(0.0f),
              800,
              48,
              512000000,
              0,
              0,
              1.01f,
              Progress.Status.Continue),
          new Progress(
              118,
              org.apache.commons.lang3.math.Fraction.getFraction(23.4f),
              -1,
              -1,
              5034667000L,
              0,
              0,
              -1,
              Progress.Status.Continue),
          new Progress(
              132,
              org.apache.commons.lang3.math.Fraction.getFraction(23.1f),
              1935500,
              1285168,
              5312000000L,
              0,
              0,
              0.929f,
              Progress.Status.End));

  public static final ImmutableList<Progress> naProgresses =
      ImmutableList.of(
          new Progress(
              0,
              org.apache.commons.lang3.math.Fraction.getFraction(0.0f),
              -1,
              -1,
              -1,
              0,
              0,
              -1,
              Progress.Status.End));
}

File: src/test/kotlin/net/bramp/ffmpeg/fixtures/Progresses.kt
package net.bramp.ffmpeg.fixtures

import net.bramp.ffmpeg.progress.Progress
import org.apache.commons.lang3.math.Fraction

val allFiles = listOf("ffmpeg-progress-0", "ffmpeg-progress-1", "ffmpeg-progress-2")

val naProgressFile = listOf("ffmpeg-progress-na")

val allProgresses = listOf(
  Progress(
    5,
    Fraction.getFraction(0.0),
    800,
    48,
    512_000_000,
    0,
    0,
    1.01f,
    Progress.Status.Continue,
  ),
  Progress(
    118,
    Fraction.getFraction(23.4),
    -1,
    -1,
    5_034_667_000L,
    0,
    0,
    -1f,
    Progress.Status.Continue,
  ),
  Progress(
    132,
    Fraction.getFraction(23.1),
    1_935_500,
    1_285_168,
    5_312_000_000L,
    0,
    0,
    0.929f,
    Progress.Status.End,
  ),
)

val naProgresses = listOf(
  Progress(
    0,
    Fraction.getFraction(0.0),
    -1,
    -1,
    -1,
    0,
    0,
    -1f,
    Progress.Status.End,
  ),
)
